resources/type-system/inference.md

```diff
@@ -1026,15 +1026,15 @@ inference depends on the context `K`, as well as the flow analysis state._
 
 A property of expression inference, known as _soundness_, is that when an
 elaborated expression is executed, it is guaranteed either to diverge, throw an
-exception, or evaluate to a value that is an _instance satisfying_ its static
-type. _Instance satisfying_ is defined as follows: a value `v` is an instance
-satisfying type `T` iff the runtime type of `v` is a subtype of the _extension
+exception, or evaluate to a value that is an _instance implementing_ its static
+type. _Instance implementing_ is defined as follows: a value `v` is an instance
```

Comment:
I think (checking ... still think it!) that the spec uses "implements `S`" to mean "has `S` as super-interface" (or really: "is `S`-bounded", which means having `S` as super-interface or is a type variable with a bound which implements `S` (recurse as needed). Maybe also a promoted type variable whose intersection type implements `S`.)

That's different from "having as supertype" if the type is generic, so `Future<int>.value(1)` is an object with runtime type `_Future<int>` which has `Future<int>` as super-interface. It implements `Future<int>`. It does not implement `Future<Object>`, it just has that as a supertype.

So, `(Future<int>.value(1) as Future<Object>)` does not satisfy that its value's runtime type implements its static type.

For soundness, if an expression evaluates to a value, the runtime type of that value must be a _subtype_ of the (extension-type-erased) static type of the expression. Using "implements" is likely wrong.


... there it was, in the "Superinterfaces" section:
> ```latex
> When we say that a type $S$
> \IndexCustom{implements}{type!implements a type}
> another type $T$,
> this means that $T$ is a superinterface of $S$,
> or $S$ is $S_0$ bounded for some type $S_0$
> (\ref{bindingActualsToFormals}),
> and $T$ is a superinterface of $S_0$.
> ... snip implementing raw types ...
> 
> \commentary{%
> Note that this is not the same as being a subtype.
> For instance, \code{List<int>} implements \code{Iterable<int>},
> but it does not implement \code{Iterable<num>}.
> Similarly, \code{List<int>} implements \code{Iterable}.
> Also, note that when $S$ implements $T$
> where $T$ is not a subtype of \code{Null},
> $S$ cannot be a subtype of \code{Null}.%
> }
```

Reply:
Ok, I see what you mean. Thanks for the explanation. I'm going to go ahead and abandon this PR and stick with the terminology "instance satisfying type T" in `resources/type-system/inference.md`.